dev-tools/mage/crossbuild.go

```diff
@@ -179,8 +194,14 @@ func (b GolangCrossBuilder) Build() error {
 
 	mountPoint := filepath.ToSlash(filepath.Join("/go", "src", repoInfo.RootImportPath))
 	workDir := mountPoint
+	buildCmd := "build/mage-linux-amd64"
 	if repoInfo.SubDir != "" {
-		workDir = filepath.ToSlash(filepath.Join(workDir, repoInfo.SubDir))
+		if b.XPack {
+			workDir = filepath.ToSlash(filepath.Join(workDir, "x-pack", repoInfo.SubDir))
+			buildCmd = filepath.Join("..", "..", repoInfo.SubDir, buildCmd)
+		} else {
+			workDir = filepath.ToSlash(filepath.Join(workDir, repoInfo.SubDir))
```

Comment:
What about seeting this line above b.Xpack check and then only have an if that overwrites it? Would be more similar to what you do with `buildCmd`. Outcome is the same.